src/Strategy.js

```diff
@@ -0,0 +1,73 @@
+//@flow
+
+import {expandStyle} from './compiler';
+import {Spec} from './ElementPropSpec';
+import {staticStylesheetManager, dynamicStylesheetManager} from './StylesheetManager';
+
+export const resolve = ({
+  defaultClassName,
+  defaultComponent,
+  ownProps,
+}: {
+  defaultClassName: Array<string>,
+  defaultComponent: *,
+  ownProps: *,
+}) => {
+  let Component = defaultComponent;
+  const className = defaultClassName;
+  let style = ownProps.style || {};
```

Comment:
I'd prefer this to be passed as a separate argument to `resolve` function. That would indicate that `style` is treated in a custom way from other props.